detectable/src/main/java/com/synopsys/integration/detectable/detectables/cargo/transform/CargoTomlTransformer.java

```diff
@@ -0,0 +1,24 @@
+package com.synopsys.integration.detectable.detectables.cargo.transform;
+
+import java.util.Optional;
+
+import com.synopsys.integration.detectable.detectables.cargo.data.CargoToml;
+import com.synopsys.integration.detectable.detectables.cargo.data.CargoTomlPackage;
+import com.synopsys.integration.util.NameVersion;
+
+public class CargoTomlTransformer {
+    public Optional<NameVersion> findProjectNameVersion(CargoToml cargoToml) {
```

Comment:
Could this return a NameOptionalVersion? Would that make sense?

Reply:
It could... This NameVersion is for the purpose of the extraction.
If Extractions started taking NameOptionalVersions I would make that change.
Seems like a separate effort. I will use NameOptionalVersion where I can though.